### PR TITLE
Remove Alamofire

### PR DESCRIPTION
**Pull request by maurovc**
## Description

As part of some optimisation tasks on our end, we identified that [Alamofire](https://github.com/Alamofire/Alamofire) was blocking and consuming a huge part of our compilation time. After some analysis we identified 3 dependencies with this library (2 external and 1 internal). In the last few months we worked to remove the internal dependency and also started helping to remove the external ones.

This PR addresses the dependency in this library by migrating the logic to `URLSession`. After running this new branch with our CI, we've seen a 17% drop on our build times. 

While we review and iterate over this PR, we're using the proposed changes as part of our internal and dev builds to stress test it and validate all scenarios.

## Proposed Changes

- Remove [Alamofire](https://github.com/Alamofire/Alamofire) and replace it by a new class `ProcessOutRequestManager`
- Replace the retry logic from [Alamofire](https://github.com/Alamofire/Alamofire) with a custom class conforming to `URLSessionDelegate`

## Things to consider

- We left out the migration of Alamofire on the Example app. We're not sure if the maintainers of the base project want to make the new request manager public (currently is only internal). Or if we also want to change to `URLSession` there. Let us know and we can help to finish this change there also.
- We kept the `SessionDelegate` within the same file as the request manager. We could also consider following the file-per-type policy so let us know if this is desired and we can also make that change.
- We're using [Cocoapods](https://github.com/CocoaPods/CocoaPods) [1.10.1](https://github.com/CocoaPods/CocoaPods/releases/tag/1.10.1), as there's no bundler integration for this project, this is the version that was used to generate the new `Podfile.lock`. Please let us know if we need to amend this to the previos version ([1.9.1](https://github.com/CocoaPods/CocoaPods/releases/tag/1.9.1)).

### CODE REVIEW

File: ProcessOut/Classes/ProcessOutRequestManager.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    }
+  
+    func HttpRequest(route: String, method: HTTPMethod, parameters: [String: Any]?, completion: @escaping (Data?, ProcessOutException?) -> Void) {
+        guard let projectId = ProcessOut.ProjectId, let authorizationHeader = self.authorizationHeader(user: projectId, password: "") else {
+            completion(nil, ProcessOutException.MissingProjectId)
+            return
+        }
+        
+        do {
+          guard let url = NSURL(string: self.apiUrl + route) else {
```
Comment (seb-preston-cko) [inline ProcessOut/Classes/ProcessOutRequestManager.swift]: The indentation on this line appears to be inconsistent with the rest of this file. Running the Xcode re-indent tool (`ctrl + i`) on this file also modified a few other lines, such as [111](https://github.com/processout/processout-ios/pull/28/files#diff-b8f36a0e052cfc509e87d0e0165055be4c546d596f39b4537c42b6a79f06dd07R111) and [116](https://github.com/processout/processout-ios/pull/28/files#diff-b8f36a0e052cfc509e87d0e0165055be4c546d596f39b4537c42b6a79f06dd07R116). I recommend running this or a similar formatting tool on your end and committing the result to ensure consistent code formatting.

**Review by seb-preston-cko — APPROVED** [review]
Thank you for your contributions @maurovc! They are greatly appreciated. I have one small request but aside from that I'm happy to approve these changes. To answer the points you raised in the PR description:

- I don't think it's necessary to refactor the example app to use the new `ProcessOutRequestManager` class at this time, and I don't think it's sensible to make the class publicly accessible for this purpose if it isn't otherwise beneficial to do so
- The file-per-type policy isn't followed elsewhere in the codebase (like in [this file](https://github.com/processout/processout-ios/blob/fe9e2bf509d7ade83fd1d0668088fbd982ea45d2/ProcessOut/Classes/GatewayConfiguration.swift), for example) so I don't see this as an issue here
- As far as I can tell, Cocoapods 1.10 doesn't introduce any breaking changes that affect us so I support this upgrade

**Comment by maurovc** [comment]
> Thank you for your contributions @maurovc! They are greatly appreciated. I have one small request but aside from that I'm happy to approve these changes. To answer the points you raised in the PR description:
> 
> 
> 
> - I don't think it's necessary to refactor the example app to use the new `ProcessOutRequestManager` class at this time, and I don't think it's sensible to make the class publicly accessible for this purpose if it isn't otherwise beneficial to do so
> 
> - The file-per-type policy isn't followed elsewhere in the codebase (like in [this file](https://github.com/processout/processout-ios/blob/fe9e2bf509d7ade83fd1d0668088fbd982ea45d2/ProcessOut/Classes/GatewayConfiguration.swift), for example) so I don't see this as an issue here
> 
> - As far as I can tell, Cocoapods 1.10 doesn't introduce any breaking changes that affect us so I support this upgrade

Hey @seb-preston-cko! Sorry I didn't see your reply before. 

I'll be glad to fullfill your request to move the PR forward. 

What would be the desired change? Let me know and I can work on it!

Thanks!

File: ProcessOut/Classes/ProcessOutRequestManager.swift (hunk 1 above)
Comment (maurovc) [reply ProcessOut/Classes/ProcessOutRequestManager.swift]: I see now this comment. Sorry, I checked the PR from the Github iOS app a few days ago and didn't see it. Let me re-run this and push it here.

**Comment by maurovc** [comment]
Hey @seb-preston-cko I updated the indentation of the changes in `RetryPolicy` and `ProcessOutRequestManager` which had the mentioned inconsistencies. Should we merge this PR or do you have any pending checks that you'd like to do?

Thanks!

**Review by seb-preston-cko — APPROVED** [review]
Thanks again for your contributions @maurovc! These changes look good to me. We'll do our best to merge them and release a new SDK version in the coming days.

**Comment by maurovc** [comment]
Hey @seb-preston-cko do we have any news about this PR? We were reviewing our internal dependencies and wanted to double check if there's any expected release time so that we can stop using our fork. Thanks!

**Pull request merged**